haas/client/node.py

```diff
@@ -0,0 +1,121 @@
+import json
+from haas.client.base import ClientBase
+from haas.client import errors
+
+
+class Node(ClientBase):
+    """ Consists of calls to query and manipulate node related
+    objects and relations """
+
+    def list(self, is_free):
+        """ List all nodes that HIL manages """
+        url = self.object_url('nodes', is_free)
+        q = self.s.get(url)
+        if q.ok:
+            return q.json()
+        elif q.status_code == 401:
+            raise errors.AuthenticationError(
+                  "Make sure credentials match chosen authentication backend."
+                  )
+
+    def show_node(self, node_name):
+        """ Shows attributes of a given node """
+
+        self.node_name = node_name
```

Comment:
I don't think you should be adding attributes to `self` here; the `Node` class is just a namespace for methods that call the server, right? it's not a single node object.